WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductSelectorViewModel.swift

```diff
@@ -292,6 +301,18 @@ final class ProductSelectorViewModel: ObservableObject {
         initialSelectedItems = []
         selectedProductIDs = []
         selectedProductVariationIDs = []
+
+        onSelectionsCleared?()
+    }
+
+    private func clearSelectedVariations() {
+        // InitialSelectedItems contains both IDs for products and variations
+        // We're sure to clear only those that belong to variations and leave the product ones
+        let initialSelectedVariations = Set(selectedProductVariationIDs)
+        initialSelectedItems = initialSelectedItems.filter { !initialSelectedVariations.contains($0) }
+
+        selectedProductVariationIDs = []
+        onSelectionsCleared?()
```

Comment:
Calling `onSelectionsCleared` triggers clearing all selected products and variations in `EditableOrderFormViewModel`. We should have a separate closure for clearing just variations like `onSelectedVariationsCleared` which clears only selected variations on the order. This is what I'm seeing when testing clearing variations only:


https://user-images.githubusercontent.com/5533851/227876413-4c443b64-9792-4538-8470-28e6436c2d0c.mp4

Reply:
Sorry I missed this one when making the previous changes. Thanks for the suggestion! Fixed on https://github.com/woocommerce/woocommerce-ios/pull/9240/commits/2bdbd38f57b2d323f86308297d669e00ba22dfb2